test/e2e/clusterctl_upgrade.go

```diff
@@ -735,7 +735,7 @@ func matchUnstructuredLists(l1 *unstructured.UnstructuredList, l2 *unstructured.
 
 // getValueOrFallback returns the input value unless it is empty, then it returns the fallback input.
 func getValueOrFallback(value []string, fallback []string) []string {
-	if len(value) > 0 {
+	if value != nil {
```

Comment:
I think it's better to use value as soon as it's not nil. This allows folks to overwrite a value with an empty slice (as needed in our use case for runtime extension providers)